log/tool_logger.go

```diff
@@ -62,27 +65,44 @@ func (tl *ToolLogger) SetDateFormat(dateFormat string) {
 	tl.format = dateFormat
 }
 
-func (tl *ToolLogger) Logvf(minVerb int, format string, a ...interface{}) {
+func (tl *ToolLogger) Logvf(minVerb int, printLogType bool, format string, a ...interface{}) {
 	if minVerb < 0 {
 		panic("cannot set a minimum log verbosity that is less than 0")
 	}
 
-	if minVerb <= tl.verbosity {
+	logLevel := minVerb
+	if minVerb == Error || minVerb == Info || minVerb == Warn {
```

Comment:
I wanted to keep using the enums so we could keep the if clause `if logLevel <= tl.verbosity`. Since Error, Info, and Warn are all replacements for Always, which is the default log level, I figured we should make sure their values are all the same, i.e. zero. The current enum ordering is as follows: Error = 0, Debug = 1, Trace = 2, Warn = 3, and Info = 4, so the values of Warn and Info would have to be overwritten in order for the if clause to work. We shouldn't need to change the program's log verbosity after this ticket, so I don't think hardcoding this value will be a problem later on.

I thought about getting rid fo the enums and just creating a map from verbosity to log levels, e.g. `verbMap["-vv"] = ["Error", "Warn", "Info", "Debug"]`, but decided against it in the end since it seemed more complicated.

Reply:
Can we just change the default tl.verbosity? Because it's weird we have 5 different logging levels after this change, but only have the ability to set 3 verbosity levels.

Reply:
We shouldn't have 5 verbosity levels, though, right? Since Error, Warn, and Info should all be logged by default, I think they should collectively be considered as one verbosity level.

Reply:
Yes, by default they should all be logged. But should we allow users to change the verbosity to Log Error only or Warning + Error etc? Conventionally most programs allow you to change the logging verbosity for each level.

Reply:
Hmm, so my understanding was that we wanted to also simplify the verbosity level structure, which explains why we combined DebugLow and DebugHigh into one log level, Trace. I do agree, though, that there's value in being able to only log the Error logs.

The design doc currently says that we should keep the original verbosity levels, so there would be just three. @tfogo , any thoughts?

Reply:
I think what you've done is a clever workaround to prevent breaking changes. But I think Huan's right that it's weird that we have 5 log levels but can only set 3. Unfortunately we would have to make a major version bump to tools in order to let people select all 5 log levels. I think our options are:

1. Make the non-breaking change where users can only select 3 of the 5 log levels. Open a ticket to add selecting 5 log levels at a later date once we have all the details around major versions sorted out.
2. Wait until we can do a major version bump of tools and do it properly. I think we would have to remove this from the mongomirror status reporting epic if we need to wait.
3. Find a way of having different logging APIs for tools and mirror. I suspect this will cause us more trouble than it's worth.

I think it's best to go with number 2. This change isn't an external request and it's more of a nice-to-have for this epic. What do you both think?

Reply:
Yea, I agree #2 might be a better solution. We can wait till break major change before working on this

Reply:
Yeah, that sounds good to me. I'll remove this ticket from the epic and make a note of this decision in the design doc.